pkg/modules/modules.go

```diff
@@ -79,9 +83,10 @@ func ProvideService(
 	logger := log.New("modules")
 
 	dependencyMap := map[string][]string{
-		HTTPServer:              {KubernetesAPIServer},
+		CertGenerator:           {},
+		HTTPServer:              {CertGenerator},
```

Comment:
Is removing the dependency on KubernetesAPIServer here intentional?

Reply:
@DanCech yep, this was a workaround i added in #65130. i had `HTTPServer` depending on `KubernetesAPIServer` so that the grafana API HTTPS server could use the same cert that was generated for the k8s HTTPS server.

since we might want to run grafana without starting the embedded k8s API server, it seemed like it made sense to split out cert generation as a separate module.

thinking about this again today, i wonder if cert gen needs to be a dskit module? certs could be generated in `certgenerator.ProvideService` and be included as a wire dependency. the only benefit of a module i can see is that we could dynamically adjust dependencies to exclude cert generation at runtime depending on deployment mode, similar to how [loki adjusts dependencies at runtime](https://github.com/grafana/loki/blob/a2370e3af3260e3705061092c0c5e8314d2669ee/pkg/loki/loki.go#L706-L709). thoughts?